sql/sql_parse_com_rpc.cc

```diff
@@ -0,0 +1,242 @@
+#include "sql_parse_com_rpc.h"
+#include "srv_session.h"
+
+#ifndef EMBEDDED_LIBRARY
+
+static const char* RpcRoleAttr = "rpc_role";
+static const char* RpcDbAttr = "rpc_db";
+
+/*
+ * @retval
+ *  false  success to find attribute
+ * @retval
+ *  true  failed to find attribute
+ * */
+static bool check_for_attribute(THD *thd, const char *attr,
+                                std::string& value) {
+  auto entity = thd->query_attrs_map.find(attr);
+  if (entity != thd->query_attrs_map.end())
+  {
+    value = entity->second;
+    return false;
+  }
+
+  return true;
+}
+
+void update_default_session_object(std::shared_ptr<Srv_session> srv_session,
```

Comment:
Using / typedef for custom shared pointer type, pass shared pointers by const reference
```suggestion
using Srv_session_ptr = std::shared_ptr<Srv_session>
void update_default_session_object(const Srv_session_ptr& srv_session,
```
```